src/5.0_to_5.1/calculated-measures/migrateCalculatedMeasures.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  ContentRecord,
+  DataModel,
+  findContentRecords,
+  getMetaData,
+} from "@activeviam/activeui-sdk-5.0";
+import _mapKeys from "lodash/mapKeys";
+import _merge from "lodash/merge";
+import { migrateCalculatedMeasureContent } from "./migrateCalculatedMeasureContent";
+import { migrateCalculatedMeasuresInDashboards } from "./migrateCalculatedMeasuresInDashboards";
+import { migrateCalculatedMeasuresInWidgets } from "./migrateCalculatedMeasuresInWidgets";
+
+const getCalculatedMeasureName = (
+  legacyCalculatedMeasureFolder: ContentRecord,
+  id: any,
+): string[] => {
+  const { structure } = legacyCalculatedMeasureFolder.children ?? {};
+  const ids = Object.keys(
+    legacyCalculatedMeasureFolder.children?.content.children ?? {},
+  );
+  const contentRecords = structure ? findContentRecords(structure, ids) : {};
+
+  const calculatedMeasureName = JSON.parse(
+    getMetaData(contentRecords[id].node, id),
+  ).name;
+
+  return calculatedMeasureName;
+};
+
+/**
+ * Transforms the serialized definitions of all calculated measures created with ActiveUI 5.0 and used in a saved dashboard or saved widget, into ones that are natively supported by ActivePivot.
+ * Removes any query-scoped calculated measure definitions from saved dashboards and saved widgets.
+ * Mutates `contentServer`.
+ */
+export function migrateCalculatedMeasures(
+  contentServer: ContentRecord,
+  dataModel: DataModel,
+): any {
+  const legacyCalculatedMeasuresFolder =
+    contentServer.children?.ui.children?.calculated_measures;
+  const cmFolder: ContentRecord | undefined =
+    contentServer.children?.pivot.children?.entitlements.children?.cm;
+
+  const legacyCalculatedMeasureRecords: {
+    [measureName: string]: ContentRecord;
+  } = legacyCalculatedMeasuresFolder?.children
+    ? _mapKeys(
+        legacyCalculatedMeasuresFolder.children.content.children,
+        (value, key) =>
+          getCalculatedMeasureName(legacyCalculatedMeasuresFolder, key),
+      )
+    : {};
+
+  const namesOfCalculatedMeasuresToMigrate = Object.keys(
+    legacyCalculatedMeasureRecords,
+  );
+
+  const {
+    migratedWidgetsRecord,
+    measureToCubeMapping: measureToCubeMappingInWidgets,
+  } = migrateCalculatedMeasuresInWidgets(
+    contentServer.children?.ui.children?.widgets!,
+    dataModel,
+    namesOfCalculatedMeasuresToMigrate,
+  );
+
+  const {
+    migratedDashboards,
+    measureToCubeMapping: measureToCubeMappingInDashboards,
+  } = migrateCalculatedMeasuresInDashboards(
+    contentServer.children?.ui.children?.dashboards!,
+    dataModel,
+    namesOfCalculatedMeasuresToMigrate,
+  );
+
+  const measureToCubeMapping = _merge(
+    measureToCubeMappingInWidgets,
+    measureToCubeMappingInDashboards,
+  );
```

Comment:
`_merge` will not work:
![image](https://user-images.githubusercontent.com/29061952/208504648-86f43753-5a9e-4b47-af71-72389c90a3ec.png)

Instead, we want to do the union of the arrays for each entry:
```suggestion
  const measureToCubeMapping = namesOfCalculatedMeasuresToMigrate.reduce((acc, measureName) => {
  const cubes = _uniq([...(measureToCubeMappingInWidgets[measureName] ?? []), ...(measureToCubeMappingInDashboards ?? [])]);
  if(cubes.length > 0){
    acc[measureName] = cubes;
  }
  return acc;
}, {});
```